bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/zk/ZKMetadataDriverBase.java

```diff
@@ -260,6 +268,46 @@ public synchronized LedgerManagerFactory getLedgerManagerFactory()
         return lmFactory;
     }
 
+    public CompletableFuture<Void> disableHealthCheck() {
+        CompletableFuture<Void> createResult = new CompletableFuture<>();
+        try {
+            zk.create(enableHealthCheckPath, BookKeeperConstants.EMPTY_BYTE_ARRAY, acls, CreateMode.PERSISTENT);
```

Comment:
Should not this be ephemeral. So that when bookie goes away and comes back again, should come back in health check  enabled mode?

Reply:
We will restart bookie frequently while upgrading the BookKeeper cluster. Do we need disabled Health Check every time we stop Bookie？

Reply:
Here, we restart with bookie as a condition for restarting the Health Check, which, in my opinion, is not reasonable.
Consider an exception case: a bookie reconnects frequently with ZooKeeper

Reply:
The purpose of this PR is to turn off the Health Check during the BookKeeper roller-restart upgrade and to enable it again after the upgrade is complete.
Bookie coming back is not a sign that the upgrade is complete

Reply:
@pkumar-singh

Reply:
@lordcheng10 I see what you mean. Thanks.